### PR TITLE
HPCC-25476 RoxieTopology restarting too often

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x]  My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Examination of generated helm, and local testing using docker desktop

### CODE REVIEW

File: helm/hpcc/values.yaml [hunk 1]
```diff
@@ -378,6 +377,10 @@ roxie:
   #serverResources:
   #  cpu: "1"
   #  memory: "1G"
+  topoServer:
+    replicas: 1
+    traceLevel: 1
+    port: 9003
```
Comment (jakesmith) [inline helm/hpcc/values.yaml:383]: in the defaultYaml in the cpp, port is 9004.. probably should be same here?

**Review by jakesmith — COMMENTED** [review]
@richardkchapman - minors + 1 question re. port # in values

File: helm/hpcc/templates/roxie.yaml [hunk 2]
```diff
@@ -20,14 +20,30 @@ data:
...
 
+{{- define "hpcc.roxieTopoConfigMap" -}}
+apiVersion: v1
+metadata:
+  name: {{ .toponame }}-configmap
+data:
+  {{ .toponame }}.yaml: |
+    version: 1.0
+    toposerver:
+      port: {{ .topoport }}
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply helm/hpcc/templates/roxie.yaml:31]: .topoport is a computed value and has been set with a default (9004)

File: helm/hpcc/values.yaml (hunk 1 above)
Comment (richardkchapman) [reply helm/hpcc/values.yaml:383]: Oops, yes, left in for testing. I will fix

File: helm/hpcc/templates/roxie.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (richardkchapman) [reply helm/hpcc/templates/roxie.yaml:31]: {{- $_ := set $commonCtx "topoport" ($toposerver.port | int | default 9004) -}}

**Comment by richardkchapman** [comment]
@jakesmith Please re-review

File: helm/hpcc/templates/roxie.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply helm/hpcc/templates/roxie.yaml:31]: right ok

**Review by jakesmith — APPROVED** [review]
Looks good

**Pull request merged**